### ISSUE TITLE
cert_chain_fuids and client_cert_chain_fuids no longer logged

### ISSUE THREAD

**Issue by philhagen** [zeek/zeek#3182]
Documentation widely reflects that cert_chain_fuids and the client equivalent are logged in x509.log, but this is not the case.  cert_chain_fps and the client equivalent are, but these have no traceability to the certificates extracted

Documentation example: https://docs.zeek.org/en/master/logs/x509.html

These fields are invaluable when correlating log entries to extracted files and should be included.

**Comment by 0xxon** [comment]
Hi,

we actually removed the file `uid`s from the log files, because they were considered cluttering. You should be able to trace the extracted certificates using the fingerprint - by default sha256 fingerprints are used, which also should be visible for certificates in `files.log`.

**Comment by philhagen** [comment]
OK - thanks for the follow-up and I can see that making sense.  However, it appears that by default `files.log` contains md5 and sha1, but not sha256.  At least this seems to be the case for week 5.2.2 as installed via `apt` on Ubuntu 22.04.2 LTS.  (if this is a packager problem, I can report it there.

```console
$ cat policy/frameworks/files/hash-all-files.zeek
##! Perform MD5 and SHA1 hashing on all files.

@load base/files/hash

event file_new(f: fa_file)
	{
	Files::add_analyzer(f, Files::ANALYZER_MD5);
	Files::add_analyzer(f, Files::ANALYZER_SHA1);
	}
```

**Comment by philhagen** [comment]
Also, since x509 certificates are not NOT listed in `files.log` by default, this is not ideal.  Working around it for now, but there are a few steps/configuration items needed to get it all working.  Also, since the full file extraction DOES extract the certificates but they are not listed in `files.log` the whole sequence is somewhat counterintuitive.

**Comment by awelzel** [comment]
@philhagen - thanks for pointing all this out. Linking #3160 as it's tangentially in the same area and came up recently. Wonder if that could be an overarching topic for 6.1 / 6.2 - @ckreibich ?

I presume you found `X509::log_x509_in_files_log` ?

For file extraction, it may make make sense to add awareness of X509 deduplication into the base scripts so the same cert isn't extracted over and over again unless . Potentially also store them by fingerprint / sha256 instead of fuids. That opens up a few questions how one would find them again if stored in daily or hourly directories.

**Comment by philhagen** [comment]
Yes, I was able to get things logged as needed with `X509:log_x509_in_files_log` as well as an extra configuration to enable sha256 checksum generation in `files.log` as well.

Thanks for the link as well - that does look related.  De-duplication would be most welcome - I could certainly see that being a major issue at scale.
